nixos/modules/services/misc/paperless.nix

```
@@ -17,13 +16,17 @@ let
     PAPERLESS_DATA_DIR = cfg.dataDir;
     PAPERLESS_MEDIA_ROOT = cfg.mediaDir;
     PAPERLESS_CONSUMPTION_DIR = cfg.consumptionDir;
-    PAPERLESS_NLTK_DIR = nltkDir;
     PAPERLESS_THUMBNAIL_FONT_NAME = defaultFont;
     GUNICORN_CMD_ARGS = "--bind=${cfg.address}:${toString cfg.port}";
   } // optionalAttrs (config.time.timeZone != null) {
     PAPERLESS_TIME_ZONE = config.time.timeZone;
   } // optionalAttrs enableRedis {
     PAPERLESS_REDIS = "unix://${redisServer.unixSocket}";
+  } // optionalAttrs (!(cfg.settings ? PAPERLESS_ENABLE_NLTK && cfg.settings.PAPERLESS_ENABLE_NLTK)) {
```

Took me a second to parse what this was doing:

```suggestion
  } // optionalAttrs (!(cfg.settings ? PAPERLESS_ENABLE_NLTK or false)) {
```

---

And now re-reading it, this looks like the test is (somewhat) inverted?

The [documentation](https://docs.paperless-ngx.com/configuration/#PAPERLESS_ENABLE_NLTK) lists the default as `true`, and the way you wrote the test, `PAPERLESS_NLTK_DIR` is defined if:
1. The user does _not_ define a value for `PAPERLESS_ENABLE_NLTK`: this is sensible enough, since it is enabled by default.
2. The user sets `PAPERLESS_ENABLE_NLTK = false`: which seems counterproductive...

So the actual suggestion should be:

```suggestion
  } // optionalAttrs (cfg.settings.PAPERLESS_ENABLE_NLTK or true) {
```

Which should do the right thing.

---

This seems to be the same thing, but yes it's way more readable :)

---

Not quite the same, if we make a table to compare.

First one:

| `PAPERLESS_ENABLE_NLTK` user configuration | `PAPERLESS_NLTK_DIR` defined |
|-|-|
| `<none>` (implicitly enabled by `paperless` itself) | :heavy_check_mark: |
| `true` | :x: |
| `false` | :heavy_check_mark: |

Second one:

| `PAPERLESS_ENABLE_NLTK` user configuration | `PAPERLESS_NLTK_DIR` defined |
|-|-|
| `<none>` (implicitly enabled by `paperless` itself) | :heavy_check_mark: |
| `true` | :heavy_check_mark: |
| `false` | :x: |

Which is what I was explaining above.